File: src/main/java/net/vidageek/games/Game.java
package net.vidageek.games;

import java.util.Collection;

import net.vidageek.games.task.TaskWithDescription;

public interface Game {

	int size();

	TaskWithDescription task(int index);

	String getDescription();

	Collection<? extends TaskWithDescription> getTasks();

	String getName();

	boolean hasNextTask(int index);

	int nextTask(int index);
}

File: src/main/scala/net/vidageek/games/Game.scala
package net.vidageek.games

import java.util.Collection

import net.vidageek.games.task.TaskWithDescription

trait Game {

  def getSize: Int

  def task(index: Int): TaskWithDescription

  def getDescription: String

  def getTasks(): Collection[_ <: TaskWithDescription]

  def getName: String

  def hasNextTask(index: Int): Boolean

  def nextTask(index: Int): Int
}